app/Http/Controllers/AssetsController.php

```diff
@@ -1198,23 +1198,25 @@ public function getBulkCheckout()
 
     public function postBulkCheckout(Request $request)
     {
-        $this->validate($request, [
-           "assigned_to"   => 'required'
-        ]);
-
-        $user = User::find(e(Input::get('assigned_to')));
         $admin = Auth::user();
-
-        if (!$user) {
-            return redirect()->route('hardware/bulkcheckout')->withInput()->with('error', trans('admin/hardware/message.checkout.user_does_not_exist'));
+        // Find checkout to type
+        if (request('checkout_to_type')=='location') {
+            $target = Location::find(request('assigned_location'));
+        } elseif (request('checkout_to_type')=='asset') {
+            $target = Asset::find(request('assigned_asset'));
+        } elseif (request('checkout_to_type')=='user') {
+            $target = User::find(request('assigned_user'));
         }
-
         if (!is_array(Input::get('selected_assets'))) {
             return redirect()->route('hardware/bulkcheckout')->withInput()->with('error', trans('admin/hardware/message.checkout.no_assets_selected'));
         }
 
         $asset_ids = array_filter(Input::get('selected_assets'));
-
+        foreach ($asset_ids as $asset_id) {
+            if ($target->id == $asset_id) {
+                return redirect()->back()->with('error', 'You cannot check an asset out to itself.');
+            }
```

Comment:
This should check to make sure that the target is an asset, otherwise you'll see an error when checking out asset with an id of 3 to location with an id of 3.

Reply:
@lea-mink Can you update this PR with @dmeltzer's changes?

Reply:
Hello @snipe, this is done. Thanks.